Firebase/InstanceID/FIRInstanceIDCheckinStore.m

```diff
@@ -146,29 +149,28 @@ - (void)saveCheckinPreferences:(FIRInstanceIDCheckinPreferences *)preferences
 }
 
 - (void)removeCheckinPreferencesWithHandler:(void (^)(NSError *error))handler {
+  // Delete the checkin preferences plist first to avoid delay.
+  NSError *deletePlistError;
+  [self.plist deleteFile:&deletePlistError];
+  if (deletePlistError) {
```

Comment:
Nit:
I think, a safer way will be:
```
if (![self.plist deleteFile:&deletePlistError]) { ...
```
`deleteFile:` uses `[NSFileManager removeItemAtPath]` which in some cases (pretty edge cases but still) may return NO but don't set an error.

Reply:
Nice catch!